analysis/dataset_clean/fn-ref.R

```diff
@@ -0,0 +1,97 @@
+    mutate(strat_cat_region = replace_na(strat_cat_region, "Missing")) %>%
+    mutate(strat_cat_region = as.factor(strat_cat_region))
+  
+  # Set reference levels for factors ---------------------------------------------
+  print('Set reference levels for factors')
+  
+  cat_factors <- colnames(input)[grepl("_cat_",colnames(input))]
+  input[,cat_factors] <- lapply(input[,cat_factors], function(x) factor(x, ordered = FALSE))
+  
+  # Set reference level for variable: sub_cat_covidhospital -------------------
+  print('Set reference level for variable: sub_cat_covidhospital')
+  
+  input$sub_cat_covidhospital <- ordered(input$sub_cat_covidhospital, 
+                                            levels = c("non_hospitalised",
+                                                       "hospitalised",
+                                                       "no_infection"))
+  
+  # Set reference level for variable: cov_cat_ethnicity --------------------------
+  print('Set reference level for variable: cov_cat_ethnicity')
+  
+  levels(input$cov_cat_ethnicity) <- list("Missing" = "0", "White" = "1", 
+                                          "Mixed" = "2", "South Asian" = "3", 
+                                          "Black" = "4", "Other" = "5")
+  
+  input$cov_cat_ethnicity <- ordered(input$cov_cat_ethnicity, 
+                                     levels = c("White","Mixed",
+                                                "South Asian","Black",
+                                                "Other","Missing"))
+  
+  # Set reference level for variable: cov_cat_imd -------------------------------
+  print('Set reference level for variable: cov_cat_imd')
+  
+  input$cov_cat_imd <- ordered(input$cov_cat_imd, 
+                               levels = c("1 (most deprived)","2","3","4","5 (least deprived)"))
+  
+  # Set reference level for variable: strat_cat_region -----------------------------
+  print('Set reference level for variable: strat_cat_region')
+  
+  input$strat_cat_region <- relevel(input$strat_cat_region, ref = "East")
+  
+  # Set reference level for variable: cov_cat_smoking ---------------------
+  print('Set reference level for variable: cov_cat_smoking')
+  
+  levels(input$cov_cat_smoking) <- list("Ever smoker" = "E", "Missing" = "M", "Never smoker" = "N", "Current smoker" = "S")
+  
+  input$cov_cat_smoking <- ordered(input$cov_cat_smoking, levels = c("Never smoker","Ever smoker","Current smoker","Missing"))
+  
+  # Set reference level for variable: cov_cat_sex --------------------------------
+  print('Set reference level for variable: cov_cat_sex')
+  
+  levels(input$cov_cat_sex) <- list("Female" = "female", "Male" = "male")
+  
+  
+  input$cov_cat_sex <- factor(input$cov_cat_sex, 
+                              levels = c("Female", "Male", "Unknown"))
+  input$cov_cat_sex <- relevel(input$cov_cat_sex, ref = "Female")
+  
+  # Define cov_cat_age_group
```

Comment:
As agreed at meeting 04/03/25, this will be removed from this script. (Apologies again for the back and forth on this decision!)